### PR TITLE
Update comments to reflect current logic

### PR DESCRIPTION
**Pull request by cofyc**
**What type of PR is this?**
> Uncomment only one, leave it on its own line:
>
> /kind api-change
> /kind bug
/kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

Update comments to reflect current logic, and log condition reason which is useful for debugging.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by cofyc** [comment]
/assign @bsalamat

In revisiting volume binder (xref https://github.com/kubernetes/kubernetes/issues/72013), I found some comments are out of date. So I'm trying to fix it.

Another thing is to log `condition.Reason` which is very useful for debugging, because it determines whether error is unschedulable or not. So I suggest to log it.

File: pkg/scheduler/internal/queue/scheduling_queue.go [hunk 1]
```diff
@@ -400,8 +401,10 @@ func (p *PriorityQueue) backoffPod(pod *v1.Pod) {
 }
 
 // AddUnschedulableIfNotPresent does nothing if the pod is present in either
-// queue. Otherwise it adds the pod to the unschedulable queue if
-// p.receivedMoveRequest is false, and to the activeQ if p.receivedMoveRequest is true.
+// queue. If pod is unschedulable, it adds pod to unschedulable queue if
```
Comment (bsalamat) [inline pkg/scheduler/internal/queue/scheduling_queue.go]: s/either queue/any queue/

File: pkg/scheduler/internal/queue/scheduling_queue.go [hunk 2]
```diff
@@ -510,9 +513,10 @@ func isPodUpdated(oldPod, newPod *v1.Pod) bool {
 	return !reflect.DeepEqual(strip(oldPod), strip(newPod))
 }
 
-// Update updates a pod in the active queue if present. Otherwise, it removes
-// the item from the unschedulable queue and adds the updated one to the active
-// queue.
+// Update updates a pod in the active or backoff queue if present. Otherwise, it removes
+// the item from the unschedulable queue if pod is updated in a that it may
```
Comment (bsalamat) [inline pkg/scheduler/internal/queue/scheduling_queue.go]: s/a that/a way that/

File: pkg/scheduler/internal/queue/scheduling_queue.go [hunk 3]
```diff
@@ -554,7 +558,7 @@ func (p *PriorityQueue) Update(oldPod, newPod *v1.Pod) error {
 		p.unschedulableQ.addOrUpdate(newPod)
 		return nil
 	}
-	// If pod is not in any of the two queue, we put it in the active queue.
+	// If pod is not in any of queues, we put it in the active queue.
```
Comment (bsalamat) [inline pkg/scheduler/internal/queue/scheduling_queue.go]: s/queues/the queues/

**Review by bsalamat — COMMENTED** [review]
Thanks, @cofyc!

**Review by bsalamat — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**